src/components/MyNdla/CopyFolder.tsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2023-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { ListResource, MessageBox, useSnack } from '@ndla/ui';
+import { TFunction, useTranslation } from 'react-i18next';
+import { useContext, useMemo, useState } from 'react';
+import { ButtonV2 as Button, LoadingButton } from '@ndla/button';
+import { GQLFolder } from '../../graphqlTypes';
+import FolderSelect from './FolderSelect';
+import {
+  useCopySharedFolder,
+  useFolders,
+} from '../../containers/MyNdla/folderMutations';
+import { AuthContext } from '../AuthenticationContext';
+import { AddResourceContainer, ButtonRow } from './AddResourceToFolder';
+
+interface Props {
+  folder: GQLFolder;
+  onClose: () => void;
+}
+
+export const baseSharedFolder = (t: TFunction) => ({
+  id: 'shared-folder',
+  name: t('myNdla.folder.sharing.sharedFolder'),
+});
+
+const CopyFolder = ({ folder, onClose }: Props) => {
+  const [selectedFolderId, setSelectedFolderId] = useState<string | undefined>(
+    undefined,
+  );
+
+  const { examLock } = useContext(AuthContext);
+  const { t } = useTranslation();
+  const { addSnack } = useSnack();
+  const { folders, loading } = useFolders();
+  const {
+    copySharedFolder,
+    loading: copySharedFolderLoading,
+    error: copySharedFolderError,
+  } = useCopySharedFolder();
```

Comment:
Har erfart at det ofte er ryddigere å bruke dette som et objekt istedenfor å destrukturere det. `copySharedFolderMutation.loading` for eksempel. Ikke nødvendig, men ofte like greit som å aliase felter